db/version_set.cc

```diff
@@ -4738,25 +4740,26 @@ class ManifestPicker {
 
  private:
   const std::string& dbname_;
-  FileSystem* const fs_;
   // MANIFEST file names(s)
   std::vector<std::string> manifest_files_;
   std::vector<std::string>::const_iterator manifest_file_iter_;
+  const std::vector<std::string>& file_names_;
```

Comment:
`file_names_` as a member variable, can be removed, since populating `manifest_files_` is light weight and can now be done in constructor. We didn't do it in the previous implementation because we avoid doing IO in constructors.

Reply:
I lean towards leaving the `manifest_files_` population in `SeekToFirstManifest()`, since if `file_names_` is empty, we'll still need to do IO in this process.

Reply:
We can still do this without having `file_names_` as member variable.
In constructor, you pass `file_names` with which you can populate the `manifest_files_`.
- If `manifest_files_` becomes not-empty afterwards, no need to populate it in `SeekToFirstManifest`.
- If `manifest_files_` is still empty when we reach `SeekToFirstManifest`, then we do IO with a local variable `files` obtained by calling `fs->GetChildren()`.